### PR TITLE
String passing fixes

### PR DESCRIPTION
**Pull request by donaldwj**
[Short description explaining the high-level reason for the pull request]

## Additions

-

## Removals

-

## Changes
Change frequently called function to take std::string& instead of std::string to avoid copys
-

## Testing

1. Auto test passed on USC7

## Screenshots
image

## Notes

-

## Todos

-

## Checklist

- [X] PR has an informative and human-readable title
- [X] Changes are limited to a single goal (no scope creep)
- [X] Code can be automatically merged (no conflicts)
- [X] Code follows project standards (link if applicable)
- [X] Passes all existing automated tests
- [X] Any _change_ in functionality is tested
- [X] New functions are documented (with a description, list of inputs, and expected output)
- [X] Placeholder code is flagged / future todos are captured in comments
- [X] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [X] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

## Testing checklist (automated report can be put here)

1.

### Target Environment support

- [ ] Linux

### CODE REVIEW

File: src/NGen.cpp [hunk 1]
```diff
@@ -69,6 +69,8 @@ int main(int argc, char *argv[]) {
               << ngen_VERSION_PATCH << std::endl;
     std::ios::sync_with_stdio(false);
 
+    ut_set_error_message_handler(ut_ignore);
```
Comment (hellkite500) [inline src/NGen.cpp:72]: need to remove this.  If you want to disable unit messages, build with `-DUDUNITS_QUIET`

**Review by hellkite500 — CHANGES_REQUESTED** [review]
Other than the unit converter message line, everything else looks like solid string optimizations.

**Comment by hellkite500** [comment]
I thought there was a larger issue about string handling for performance, but I'm not finding it, but I did want to note the relation to #443 and #444.

File: src/NGen.cpp (hunk 1 above)
Comment (donaldwj) [reply src/NGen.cpp:72]: I think that is a commit that somehow got from one of my profiling branches into my main and has been rebased onto code sense then.

Comment (donaldwj) [reply src/NGen.cpp]: Any it is removed

**Review by hellkite500 — APPROVED** [review]

**Comment by donaldwj** [comment]
String Fixes pull request #520

**Pull request merged**